modules/tensor_mechanics/src/actions/MechanicalBodyForceAction.C

```diff
@@ -0,0 +1,56 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MechanicalBodyForceAction.h"
+#include "Factory.h"
+#include "FEProblem.h"
+#include "Conversion.h"
+
+registerMooseAction("TensorMechanicsApp", MechanicalBodyForceAction, "add_bc");
+
+InputParameters
+MechanicalBodyForceAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Set up Pressure boundary conditions");
+
+  params.addParam<std::vector<SubdomainName>>("block",
+                                              "The block ids where the body force will be applied");
+
+  params.addParam<std::vector<VariableName>>(
+      "displacements",
+      "The displacements appropriate for the simulation geometry and coordinate system");
+
+  params.addParam<FunctionName>(
+      "function", "1", "Function to scale the coupled body force vector property");
+  params.addParam<Real>(
+      "hht_alpha", 0.0, "alpha parameter required for HHT time integration scheme");
+  params.addRequiredParam<MaterialPropertyName>("force_density", "Force per unit volume vector");
```

Comment:
I think my preference would be to just call it 'body_force'. Regardless of what we call it, it should be consistent between the action and Kernel, which is not the case right now.